File: src/main/java/com/kapilkoju/nepse/data/todaysprice/TodaysPriceController.java
package com.kapilkoju.nepse.data.todaysprice;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;
import reactor.core.publisher.Flux;

@RestController
public class TodaysPriceController {

    private final TodaysPriceService todaysPriceService;

    public TodaysPriceController(TodaysPriceService todaysPriceService) {
        this.todaysPriceService = todaysPriceService;
    }

    @GetMapping("/data/todaysprice")
    public Flux<TodaysPriceEntry> getTodaysPrice() {
        return todaysPriceService.getTodaysPrice();
    }
}

File: src/main/kotlin/com/kapilkoju/nepse/data/todaysprice/TodaysPriceController.kt
package com.kapilkoju.nepse.data.todaysprice

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@RestController
class TodaysPriceController(private val todaysPriceService: TodaysPriceService) {

    @GetMapping("/data/todaysprice")

    fun getTodaysPrice(): List<TodaysPriceEntry>
            = todaysPriceService.getTodaysPrice()
}
